dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/command/handler/AbstractCommandHandler.java

```diff
@@ -106,6 +111,10 @@ protected void assembleWorkflowDefinition(
         checkArgument(workflowDefinition != null,
                 "Cannot find the WorkflowDefinition: [" + workflowDefinitionCode + ":" + workflowDefinitionVersion
                         + "]");
+        Project project = projectDao.queryByCode(workflowDefinition.getProjectCode());
+        if (project != null) {
+            workflowDefinition.setProjectName(project.getName());
+        }
```

Comment:
`WorkflowDefinition` is a DO, please don't add field to this which doesn't exist in db, this will make it hard to judge which and when it is not null.

It's better to assemble project into `WorkflowExecuteContext`.

Reply:
> `WorkflowDefinition` is a DO, please don't add field to this which doesn't exist in db, this will make it hard to judge which and when it is not null.
> 
> It's better to assemble project into `WorkflowExecuteContext`.

done